packages/host/app/services/local-realm.ts

```diff
@@ -48,19 +62,67 @@ export default class LocalRealm extends Service {
         }
         break;
       case 'available':
-        if (data.type === 'visitRequest') {
-          let { id, path, staticResponses } = data;
+        if (data.type === 'setEntryAcknowledged') {
+          if (!this.#setEntryDeferred) {
+            throw new Error(
+              `received setEntry acknowledgement without corresponding setEntry request`
+            );
+          }
+          this.#setEntryDeferred.fulfill();
+          return;
+        }
+        if (data.type === 'startFromScratch') {
+          if (!this.#fromScratch) {
+            throw new Error(
+              `the fromScratch runner has not been registered with the local realm service`
+            );
+          }
+          let { realmURL } = data;
+          let worker = this.state.worker;
+          this.#fromScratch(new URL(realmURL)).then((state) => {
+            send(worker, {
+              type: 'fromScratchCompleted',
+              state: serializeRunState(state),
```

Comment:
it seems wasteful to have to bother with serializing and sending the state.instances here since setEntries built it up.